src/index.js

```diff
@@ -0,0 +1,131 @@
+      },
+    })
+    .then((response) => {
+      console.log(response.data);
+      latitude = response.data[0].lat;
+      longitude = response.data[0].lon;
+      console.log('success in findLatitudeAndLongitude!', latitude, longitude);
+      getWeather(latitude, longitude);
+    })
+    .catch((error) => {
+      console.log('error in findLatitudeAndLongitude!', error);
+    });
+};
+// -------------- Wave 5 --------------------
+const getSkyChoice = () => {
+  const skyDropdown = document.querySelector('select');
+  const skyChoice = skyDropdown.value;
+  console.log('sky choice:', skyChoice);
+  state.sky = skyChoice;
+  if (state.sky === 'sunny') {
+    document.getElementById('sky-emojis').innerText = '☁️ ☁️ ☁️ ☀️ ☁️ ☁️';
+  } else if (state.sky === 'cloudy') {
+    document.getElementById('sky-emojis').innerText =
+      '☁️☁️ ☁️ ☁️☁️ ☁️ 🌤 ☁️ ☁️☁️';
+  } else if (state.sky === 'rainy') {
+    document.getElementById('sky-emojis').innerText = '🌧🌈⛈🌧🌧💧⛈🌧🌦🌧💧🌧🌧';
+  } else if (state.sky === 'snowy') {
+    document.getElementById('sky-emojis').innerText = '🌨❄️🌨🌨❄️❄️🌨❄️🌨❄️❄️🌨🌨';
+  }
+};
+// getting temperature from latitude and longitude
+const getWeather = (latitude, longitude) => {
+  axios
+    .get('http://127.0.0.1:5000/weather', {
+      params: { lat: latitude, lon: longitude },
+    })
+    .then((response) => {
+      console.log('success in getWeather!', response.data.main.temp);
+      let kelvin = response.data.main.temp;
+      let fahrenheit = Math.round(((kelvin - 273.15) * 9) / 5 + 32);
+      state.temp = fahrenheit;
+      const tempContainer = document.querySelector('#degrees');
+      tempContainer.textContent = state.temp;
+      updateColorsAndEmojis();
+    })
+    .catch((error) => {
+      console.log('error in getWeather!');
+    });
+};
+// ...Wave 6..............
+const resetFunc = () => {
+  document.querySelector('#search-bar').value = '';
+  updateCity();
+  const tempContainer = document.querySelector('#degrees');
+  tempContainer.textContent = 0;
+};
+// registering event handlers
+const registerHandlers = (event) => {
+  const downArrow = document.querySelector('#down-arrow');
+  downArrow.addEventListener('click', subtractDegree);
+  const upArrow = document.querySelector('#up-arrow');
+  upArrow.addEventListener('click', addDegree);
+  const changeTempColor = document.querySelector('#degrees');
+  const updateCityName = document.querySelector('#search-bar');
+  updateCityName.addEventListener('input', updateCity);
+  changeTempColor.addEventListener('click', updateColorsAndEmojis);
+  const tempButton = document.querySelector('#get-temp');
+  tempButton.addEventListener('click', findLatitudeAndLongitude);
+  const updateSky = document.querySelector('select');
+  updateSky.addEventListener('change', getSkyChoice);
+  const resetCity = document.querySelector('#reset');
+  resetCity.addEventListener('click', resetFunc);
+};
+document.addEventListener('DOMContentLoaded', registerHandlers);
```

Comment:
Consider adding the more robust page loading handling seen at the bottom of the crab fan site example.